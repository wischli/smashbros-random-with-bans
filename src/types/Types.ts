export interface IChar {
  id: number;
  name: string;
  enabled: boolean;
  played: boolean;
  display: boolean;
  echo: number[];
  media: string;
}

export interface ICState {
  enabled: number[];
  played: number[];
  disabled: number[];
  hidden: number[];
  date: string;
}

export interface IState {
  enabled: number[];
  played: number[];
  disabled: number[];
  hidden: number[];
}

export interface ICookies {
  characters: ICState | boolean;
  notice: boolean;
}
