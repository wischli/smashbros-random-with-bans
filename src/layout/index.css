body {
  margin: 0;
  padding: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
    monospace;
}

.close {
  opacity: 1;
}
.close:hover {
  opacity: 1;
  background-color: #ffb3b3 !important;
  transition: background ease-in-out .5s;
}
.close:before, .close:after {
  position: absolute;
  left: 13px;
  content: ' ';
  height: 19px;
  width: 2px;
  top: 4px;
  background-color: white;
}
.close2:before, .close2:after {
  position: absolute;
  left: 23px;
  content: ' ';
  height: 33px;
  width: 2px;
  top: 8px;
  background-color: white;
}
.close:before {
  transform: rotate(45deg);
}
.close:after {
  transform: rotate(-45deg);
}
