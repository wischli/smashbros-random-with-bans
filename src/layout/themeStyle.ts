export const color = {
  background: '#486471',
  bgContent: '#486471',
  red: '#ffb3b3',
  green: '#99f9ae',
  card: '#26353c',
  cardTitle: '#aacdde',
  nav: '#466370',
  bgNavSecBtn: '#ddddddb3',
  bgCookieNotice: '#26353c',
  charEnabled: 'white',
  charDisabled: '#ff6767b8',
  charPlayed: '#70f78e',
  charHidden: 'grey',
  cardBtnLeft: '#383838',
  cardBtnRight: 'black',
};

export const themeButtonStyle: React.CSSProperties = {
  height: '50px',
  fontSize: '15px',
  fontWeight: 700,
  textAlign: 'center',
  justifyContent: 'center',
  borderRadius: '5px',
  textTransform: 'uppercase',
  letterSpacing: '1px',
  border: '2px solid',
};
