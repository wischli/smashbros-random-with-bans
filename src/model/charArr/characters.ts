export const characters: {
  id: number,
  name: string,
  echo: number[],
}[] = [
    {
    id: 1,
    name: 'Mario',
    echo: [],
  },
  {
    id: 43,
    name: 'Dr. Mario',
    echo: [],
  },
  {
    id: 2,
    name: 'Luigi',
    echo: [],
  },
  {
    id: 3,
    name: 'Peach',
    echo: [3.5],
  },
  {
    id: 3.5,
    name: 'Daisy',
    echo: [3],
  },
  {
    id: 4,
    name: 'Bowser',
    echo: [],
  },
  {
    id: 5,
    name: 'Yoshi',
    echo: [],
  },
  {
    id: 6,
    name: 'Rosalina & Luma',
    echo: [],
  },
  {
    id: 7,
    name: 'Donkey Kong',
    echo: [],
  },
  {
    id: 8,
    name: 'Diddy Kong',
    echo: [],
  },
  {
    id: 9,
    name: 'Link',
    echo: [],
  },
  {
    id: 10,
    name: 'Zelda',
    echo: [],
  },
  {
    id: 11,
    name: 'Sheik',
    echo: [],
  },
  {
    id: 45,
    name: 'Ganondorf',
    echo: [],
  },
  {
    id: 12,
    name: 'Toon Link',
    echo: [],
  },
  {
    id: 12.9,
    name: 'Young Link',
    echo: [],
  },
  {
    id: 13,
    name: 'Samus',
    echo: [13.5],
  },
  {
    id: 13.5,
    name: 'Dark Samus',
    echo: [13],
  },
  {
    id: 14,
    name: 'Zero Suit Samus',
    echo: [],
  },
  {
    id: 15,
    name: 'Pit',
    echo: [15],
  },
  {
    id: 15.5,
    name: 'Dark Pit',
    echo: [15.5],
  },
  {
    id: 16,
    name: 'Palutena',
    echo: [],
  },
  {
    id: 17,
    name: 'Marth',
    echo: [17.5],
  },
  {
    id: 17.5,
    name: 'Lucina',
    echo: [17],
  },
  {
    id: 18,
    name: 'Ike',
    echo: [],
  },
  {
    id: 19,
    name: 'Robin',
    echo: [],
  },
  {
    id: 20,
    name: 'Kirby',
    echo: [],
  },
  {
    id: 21,
    name: 'King Dedede',
    echo: [],
  },
  {
    id: 22,
    name: 'Meta Knight',
    echo: [],
  },
  {
    id: 23,
    name: 'Little Mac',
    echo: [],
  },
  {
    id: 24,
    name: 'Fox',
    echo: [],
  },
  {
    id: 25,
    name: 'Pikachu',
    echo: [],
  },
  {
    id: 25.9,
    name: 'Pichu',
    echo: [],
  },
  {
    id: 26,
    name: 'Pokemon Trainer',
    echo: [],
  },
  {
    id: 27,
    name: 'Lucario',
    echo: [],
  },
  {
    id: 28,
    name: 'Greninja',
    echo: [],
  },
  {
    id: 29,
    name: 'Captain Falcon',
    echo: [],
  },
  {
    id: 30,
    name: 'Villager',
    echo: [],
  },
  {
    id: 31,
    name: 'Olimar',
    echo: [],
  },
  {
    id: 32,
    name: 'Wii Fit Trainer',
    echo: [],
  },
  {
    id: 33,
    name: 'Shulk',
    echo: [],
  },
  {
    id: 34,
    name: 'Pac-Man',
    echo: [],
  },
  {
    id: 35,
    name: 'Mega Man',
    echo: [],
  },
  {
    id: 36,
    name: 'Sonic',
    echo: [],
  },
  {
    id: 38,
    name: 'Ness',
    echo: [],
  },
  {
    id: 39,
    name: 'Falco',
    echo: [],
  },
  {
    id: 40,
    name: 'Wario',
    echo: [],
  },
  {
    id: 41,
    name: 'Snake',
    echo: [],
  },
  {
    id: 42,
    name: 'Ice Climbers',
    echo: [],
  },
  {
    id: 44,
    name: 'R.O.B.',
    echo: [],
  },
  {
    id: 46,
    name: 'Mr. Game & Watch',
    echo: [],
  },
  {
    id: 47,
    name: 'Bowser Jr.',
    echo: [],
  },
  {
    id: 48,
    name: 'Duck Hunt',
    echo: [],
  },
  {
    id: 49,
    name: 'Jigglypuff',
    echo: [],
  },
  {
    id: 50,
    name: 'Mewtwo',
    echo: [],
  },
  {
    id: 51,
    name: 'Lucas',
    echo: [],
  },
  {
    id: 52,
    name: 'Roy',
    echo: [52.5],
  },
  {
    id: 52.5,
    name: 'Chrom',
    echo: [52],
  },
  {
    id: 53,
    name: 'Ryu',
    echo: [53.5],
  },
  {
    id: 53.5,
    name: 'Ken',
    echo: [53],
  },
  {
    id: 54,
    name: 'Cloud',
    echo: [],
  },
  {
    id: 55,
    name: 'Corrin',
    echo: [],
  },
  {
    id: 56,
    name: 'Bayonetta',
    echo: [],
  },
  {
    id: 57,
    name: 'Mii Fighter Brawler',
    echo: [],
  },
  {
    id: 58,
    name: 'Mii Fighter Shooter',
    echo: [],
  },
  {
    id: 59,
    name: 'Mii Fighter Sword',
    echo: [],
  },
  {
    id: 60,
    name: 'Inkling',
    echo: [],
  },
  {
    id: 61,
    name: 'Ridley',
    echo: [],
  },
  {
    id: 62,
    name: 'Simon',
    echo: [62.5],
  },
  {
    id: 62.5,
    name: 'Richter',
    echo: [62],
  },
  {
    id: 63,
    name: 'King K. Rool',
    echo: [],
  },
  {
    id: 64,
    name: 'Isabelle',
    echo: [],
  },
  {
    id: 65,
    name: 'Incineroar',
    echo: [],
  },
  {
    id: 66,
    name: 'Piranha Plant',
    echo: [],
  },
];
