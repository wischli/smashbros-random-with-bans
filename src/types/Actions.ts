export enum Action {
  randomize = 'randomize',
  next = 'next',
  previous = 'previous',
  restore = 'restore',
  echo = 'echo',
  save = 'save',
  reset = 'reset',
  toggleChar = 'toggleChar',
}
