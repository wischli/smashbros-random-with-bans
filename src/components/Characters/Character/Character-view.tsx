import React from 'react';
import { IChar, IState } from '../../../types/Types';
import { charStyle, imageStyle } from './Character-style';

export const Character = (props: { character: IChar; stateKey: keyof IState; charIndex: number; handleCharClick: Function }) => {
  const { character, charIndex, stateKey, handleCharClick } = props;
  const getClassName = (played: boolean, enabled: boolean): keyof Omit<IState, 'hidden'> => {
    if (played) {
      return 'played';
    }
    return enabled ? 'enabled' : 'disabled';
  };
  return (
    <div
      className="character"
      role="button"
      tabIndex={0}
      id={`${character.id}`}
      onClick={() => handleCharClick(charIndex, stateKey)}
      onKeyPress={() => handleCharClick(charIndex, stateKey)}
      style={charStyle(stateKey)}
    >
      <img src={character.media} style={imageStyle(stateKey)} alt={character.name} className={getClassName(character.played, character.enabled) as string} />
    </div>
  );
};
