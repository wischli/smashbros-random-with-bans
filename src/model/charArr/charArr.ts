import { IChar, IState } from '../../types/Types'
import { characters } from './characters'
import { imigify } from '../../utils'

export const charArr: IChar[] = characters.map(char => ({
  ...char,
  media: imigify(char.name),
  enabled: true,
  played: false,
  display: true,
}))

export const initialCharState: IState = { enabled: charArr.map((_, index) => index), played: [], disabled: [], hidden: [] }
