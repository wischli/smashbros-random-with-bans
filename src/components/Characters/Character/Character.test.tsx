/* eslint-disable no-undef */

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Character } from './Character-view';
import { color } from '../../../layout/themeStyle';
import { charArr } from '../../../model/charArr/charArr';
import { IChar } from '../../../types/Types';

configure({ adapter: new Adapter() });

describe('Testing Character Component', () => {
  // set default test data
  const context = {
    handleCharClick: jest.fn(),
  };
  let testChar: IChar;
  beforeEach(() => {
    [testChar] = charArr;
  });

  it('Checks enabled character', () => {
    const wrapper = shallow(<Character character={testChar} stateKey="enabled" charIndex={0} handleCharClick={context.handleCharClick} />);
    try {
      expect(wrapper.get(0).props.children.props.style.opacity).toEqual(1);
      expect(wrapper.get(0).props.style.backgroundColor).toBe(color.charEnabled);
    } catch (err) {
      expect(err).toBeUndefined();
    }
    expect(wrapper).toMatchSnapshot();
  });

  it('Checks disabled character', () => {
    const wrapper = shallow(<Character character={testChar} stateKey="disabled" charIndex={0} handleCharClick={context.handleCharClick} />);
    try {
      expect(wrapper.get(0).props.children.props.style.opacity).toEqual(0.6);
      expect(wrapper.get(0).props.style.backgroundColor).toBe(color.charDisabled);
    } catch (err) {
      expect(err).toBeUndefined();
    }
    expect(wrapper).toMatchSnapshot();
  });

  it('Checks played character', () => {
    const wrapper = shallow(<Character character={testChar} stateKey="played" charIndex={0} handleCharClick={context.handleCharClick} />);
    try {
      expect(wrapper.get(0).props.children.props.style.opacity).toEqual(0.6);
      expect(wrapper.get(0).props.style.backgroundColor).toBe(color.charPlayed);
    } catch (err) {
      expect(err).toBeUndefined();
    }
    expect(wrapper).toMatchSnapshot();
  });

  it('Checks hidden character', () => {
    const wrapper = shallow(<Character character={testChar} stateKey="hidden" charIndex={0} handleCharClick={context.handleCharClick} />);
    try {
      expect(wrapper.get(0).props.children.props.style.opacity).toEqual(0.6);
      expect(wrapper.get(0).props.style.backgroundColor).toBe(color.charHidden);
    } catch (err) {
      expect(err).toBeUndefined();
    }
    expect(wrapper).toMatchSnapshot();
  });
});
