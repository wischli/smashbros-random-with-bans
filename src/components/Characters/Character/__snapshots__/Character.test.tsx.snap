// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Character Component Checks disabled character 1`] = `
<div
  className="character"
  id="1"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "backgroundColor": "#ff6767b8",
      "borderRadius": "100%",
      "display": "block,",
      "maxHeight": 85,
      "maxWidth": 85,
    }
  }
  tabIndex={0}
>
  <img
    alt="Mario"
    className="enabled"
    src="http://localhost/public/images/character_icons/mario.png"
    style={
      Object {
        "display": "block",
        "margin": "auto",
        "maxWidth": 80,
        "opacity": 0.6,
        "padding": 1,
      }
    }
  />
</div>
`;

exports[`Testing Character Component Checks enabled character 1`] = `
<div
  className="character"
  id="1"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": "100%",
      "display": "block,",
      "maxHeight": 85,
      "maxWidth": 85,
    }
  }
  tabIndex={0}
>
  <img
    alt="Mario"
    className="enabled"
    src="http://localhost/public/images/character_icons/mario.png"
    style={
      Object {
        "display": "block",
        "margin": "auto",
        "maxWidth": 80,
        "opacity": 1,
        "padding": 1,
      }
    }
  />
</div>
`;

exports[`Testing Character Component Checks hidden character 1`] = `
<div
  className="character"
  id="1"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "backgroundColor": "grey",
      "borderRadius": "100%",
      "display": "block,",
      "maxHeight": 85,
      "maxWidth": 85,
    }
  }
  tabIndex={0}
>
  <img
    alt="Mario"
    className="enabled"
    src="http://localhost/public/images/character_icons/mario.png"
    style={
      Object {
        "display": "block",
        "margin": "auto",
        "maxWidth": 80,
        "opacity": 0.6,
        "padding": 1,
      }
    }
  />
</div>
`;

exports[`Testing Character Component Checks played character 1`] = `
<div
  className="character"
  id="1"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "backgroundColor": "#70f78e",
      "borderRadius": "100%",
      "display": "block,",
      "maxHeight": 85,
      "maxWidth": 85,
    }
  }
  tabIndex={0}
>
  <img
    alt="Mario"
    className="enabled"
    src="http://localhost/public/images/character_icons/mario.png"
    style={
      Object {
        "display": "block",
        "margin": "auto",
        "maxWidth": 80,
        "opacity": 0.6,
        "padding": 1,
      }
    }
  />
</div>
`;
